mlir/include/mlir/Dialect/LLVMIR/LLVMOps.td

```diff
@@ -1196,7 +1197,8 @@ def LLVM_GlobalOp : LLVM_Op<"mlir.global",
       CArg<"bool", "false">:$dsoLocal,
       CArg<"bool", "false">:$thread_local_,
       CArg<"SymbolRefAttr", "{}">:$comdat,
-      CArg<"ArrayRef<NamedAttribute>", "{}">:$attrs)>
+      CArg<"ArrayRef<NamedAttribute>", "{}">:$attrs,
+      CArg<"DIGlobalVariableExpressionAttr", "{}">:$dbgExpr)>
```

Comment:
Can you explain why the debug expression is an argument of the GlobalOp?

I would expect that the global variable expression is stored as part of the location using a fused location. An example is the DISubprogram that is part of a function location. It is stored using a `FusedLocWith<LLVM::DISubprogramAttr>`. Something similar should be possible for the DIGlobalVariableExpressionAttr.

Reply:
This is criticism I was waiting for 😄. I had reservations about adding this as a parameter directly to GlobalOp as this is not scoping information, like DICompileUnit, DILexicalBlock or DISubprogram, that implies an location. The docs clearly says `Debug scoping information is attached to LLVM IR dialect operations using a fused location (FusedLoc) whose metadata holds the DIScopeAttr representing the debug scope`, so I figured this needed to be different.

Reply:
That is a good argument point taken :).